app/controllers/campaign_budgets_controller.rb

```diff
@@ -0,0 +1,14 @@
+class CampaignBudgetsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_campaign
+
+  def show
+  end
```

Comment:
I've changed my thinking on having empty actions in controllers. While it does communicate some intent, the routes file is the true canonical source of what actions exist. I see these empty method defs as unnecessary noise... at least for experienced Rails developers. This practice could lead to cruft that we forget to clean up if/when the routes change.

No need to remove right now. I just wanted to communicate my thoughts on this pattern.

Reply:
Can you help me understand this better? How would you handle this?

Reply:
I'd just omit the empty method. Rails will simply render view template if the route is defined.

Reply:
In light of the fact that this is open source, do you think that it may be confusing to less experienced Rails developers?